tests/phpunit/tests/image/editor.php

```diff
@@ -363,4 +363,88 @@ public function data_wp_get_webp_info() {
 			),
 		);
 	}
+
+	/**
+	 * Test wp_get_avif_info.
+	 *
+	 * @ticket 51228
+	 * @dataProvider data_wp_get_avif_info
+	 *
+	 */
+	public function test_wp_get_avif_info( $file, $expected ) {
+		$editor = wp_get_image_editor( $file );
+
+		if ( is_wp_error( $editor ) || ! $editor->supports_mime_type( 'image/avif' ) ) {
+			$this->markTestSkipped( sprintf( 'No AVIF support in the editor engine %s on this system.', $this->editor_engine ) );
+		}
+
+		$file_data = wp_get_avif_info( $file );
+		$this->assertSame( $expected, $file_data );
+	}
+
+	/**
+	 * Data provider for test_wp_get_avif_info().
+	 */
+	public function data_wp_get_avif_info() {
+		return array(
+			// Standard JPEG.
+			array(
+				DIR_TESTDATA . '/images/test-image.jpg',
+				array(
+					'width'  => false,
+					'height' => false,
+					'type'   => false,
+				),
+			),
+			// Standard GIF.
+			array(
+				DIR_TESTDATA . '/images/test-image.gif',
+				array(
+					'width'  => false,
+					'height' => false,
+					'type'   => false,
+				),
+			),
+			// Animated AVIF.
+			array(
+				DIR_TESTDATA . '/images/avif-animated.avif',
+				array(
+					'width'        => 150,
+					'height'       => 150,
+					'bit_depth'    => 8,
+					'num_channels' => 3,
+				),
+			),
+			// Lossless AVIF.
+			array(
+				DIR_TESTDATA . '/images/avif-lossless.avif',
+				array(
+					'width'        => 400,
+					'height'       => 400,
+					'bit_depth'    => 8,
+					'num_channels' => 3,
+				),
+			),
+			// Lossy AVIF.
+			array(
+				DIR_TESTDATA . '/images/avif-lossy.avif',
+				array(
+					'width'        => 400,
+					'height'       => 400,
+					'bit_depth'    => 8,
+					'num_channels' => 3,
+				),
+			),
+			// Transparent AVIF.
+			array(
+				DIR_TESTDATA . '/images/avif-transparent.avif',
+				array(
+					'width'        => 128,
+					'height'       => 128,
+					'bit_depth'    => 12,
+					'num_channels' => 3,
```

Comment:
This should be 4 if the AVIF is translucent.

Reply:
Apparently the image I used to the tests wasn't actually transparent. Once I correct the image, this test should fail - then I will fix the test.

Reply:
I swapped in the gradient image you shared and the test still passes with 3 channels at least locally, running the full matrix on the PR now. I added a few more test images from libavif and also made sure the tests for `wp_get_avif_info` always run regardless of server support for AVIF (also for the WebP tests).

> This should be 4 if the AVIF is translucent.

I think the image is right, so maybe there is something wrong with the PHP code? This isn't critical since we don't actually use the `bit_depth` anywhere, however it would be good to dig in further to see why this isn't working.

Reply:
> alpha string comparison was off in PHP

Well, the good news is running the tests here caught the bug! I will apply the changes you suggested and update the tests.

Reply:
Fixed in [`b7c193c` (#4612)](https://github.com/WordPress/wordpress-develop/pull/4612/commits/b7c193ca55470bc50eac13ee3f05c7d17ead0d87)